src/ddmd/astbasevisitor.d

```diff
@@ -0,0 +1,1071 @@
+module ddmd.astbasevisitor;
+
+import ddmd.astbase;
+
+class Visitor
+{
+    void visit(ASTBase.Dsymbol)
```

Comment:
There are a bunch of repetitive functions that are one-liners. I suggest you take the license at least for the code you introduce to write them one per line without separating whitespace, e.g.:

```d
void visit(ASTBase.Dsymbol) { assert(0); }
void visit(ASTBase.Dsymbol) { visit(cast(ASTBase.Dsymbol)s); }
void visit(ASTBase.Declaration s) { visit(cast(ASTBase.Dsymbol)s); }
...
```

We're looking at literally 5x LOC compression over large numbers of lines. That helps reviewers and contributors wading through code and also make certain patterns easier to see.